pool/pool.go

```diff
@@ -91,6 +95,9 @@ type clientStatus interface {
 	methodsStatus() []statusSnapshot
 }
 
+// ErrPoolClientUnhealthy is an error to indicate that client in pool is unhealthy.
+var ErrPoolClientUnhealthy = errors.New("pool client unhealthy")
```

Comment:
Do we need to handle this error outside the package? If not, do not export it.

Reply:
In theory we can do retry in app code if  such error happened

Reply:
We should either:
* wait when theory becomes reality, or
* mention this error in docs of all methods which can return it